core/src/wsv.rs

```diff
@@ -848,6 +883,21 @@ impl WorldStateView {
             .collect::<Vec<Parameter>>()
     }
 
+    /// Query parameter and convert it to proper type
+    pub fn query_param<T: TryFrom<Value>, P: core::hash::Hash + Eq + ?Sized>(
+        &self,
+        param: &P,
+    ) -> Option<T>
+    where
+        Parameter: Borrow<P>,
+    {
+        self.world
+            .parameters
+            .get(param)
+            .as_ref()
+            .and_then(|param| param.key().val.clone().try_into().ok())
+    }
```

Comment:
As our parameter names and types aren't defined at runtime I wonder if it makes sence to hardcode them into a structure? With that it would be possible to remove a lot of usages like: 

```rust
wsv.query_param::<u32>("OopsIMadeATypoHere")
```

Reply:
@Arjentix I'm not quite follow your idea, can you explain it a bit more?

Reply:
What about having this:

```rust
pub struct Parameters 
    param_1: Type1,
    // ...
}
```

Why do we need to have `ParametersMap`?

Reply:
I suggest to hardcode it like we do it in `cli/config`

Reply:
Oh, i see, i [supposed](https://github.com/hyperledger/iroha/pull/2973#discussion_r1033160209) this idea when `Parameter`s was firstly introduced. 

But we stick with `DashMap` because it give us more flexibility afaik. I mean this way we can store parameters that will became useful later or don't cause trouble if some parameters will be removed.

Reply:
I don't quite understand this flexibility. We can use previously unused parameters only by upgrading Iroha code, so I can't see a reason to set such parameters before that. Also unused parameters may indicate an user error, because they might use outdated ones

Reply:
Okay, I see this discussions in your comment

Reply:
Let's leave it as it is then